routes/weatherProviders/DarkSky.ts

```diff
@@ -105,9 +107,61 @@ async function getDarkSkyWeatherData( coordinates: GeoCoordinates ): Promise< We
 	return weather;
 }
 
+async function getDarkSkyEToData( coordinates: GeoCoordinates ): Promise< EToData > {
+	// TODO use a rolling 24 hour window instead of fixed calendar days?
+	// The Unix epoch seconds timestamp of 24 hours ago.
+	const timestamp: number = moment().subtract( 1, "day" ).unix();
+
+	const DARKSKY_API_KEY = process.env.DARKSKY_API_KEY,
+		historicUrl = `https://api.darksky.net/forecast/${DARKSKY_API_KEY}/${coordinates[0]},${coordinates[1]},${timestamp}?units=si`;
+
+	let historicData;
+	try {
+		historicData = await httpJSONRequest( historicUrl );
+	} catch (err) {
+		// Indicate ETO data could not be retrieved if an API error occurs.
+		return undefined;
+	}
+
+	// Approximate solar radiation using a formula from http://www.shodor.org/os411/courses/_master/tools/calculators/solarrad/
+	const solarRadiation = historicData.hourly.data.reduce( ( total, hour ) => {
+		const currPosition = SunCalc.getPosition( new Date( hour.time * 1000 ), coordinates[ 0 ], coordinates[ 1 ] );
+		// The Sun's position 1 hour ago.
+		const lastPosition = SunCalc.getPosition( new Date( hour.time * 1000 - 60 * 60 * 1000 ), coordinates[ 0 ], coordinates[ 1 ] );
+		const solarElevationAngle = ( currPosition.altitude + lastPosition.altitude ) / 2;
+
+		// Calculate radiation and convert from watts to megajoules.
+		const clearSkyIsolation = ( 990 * Math.sin( solarElevationAngle ) - 30 ) * 0.0036;
+
+		// TODO include the radiation from hours where the sun is partially above the horizon.
+		// Skip hours where the sun was below the horizon.
+		if ( clearSkyIsolation <= 0 ) {
+			return total;
+		}
+
+		return total + clearSkyIsolation * ( 1 - 0.75 * Math.pow( hour.cloudCover, 3.4 ) );
+	}, 0 );
+
+	return {
+		weatherProvider: "DarkSky",
+		minTemp: historicData.daily.data[ 0 ].temperatureMin,
+		maxTemp: historicData.daily.data[ 0 ].temperatureMax,
+		minHumidity: historicData.hourly.data.reduce( ( min, hour ) => Math.min( min, hour.humidity ), 1) * 100,
+		maxHumidity: historicData.hourly.data.reduce( ( max, hour ) => Math.max( max, hour.humidity ), 0) * 100,
+		solarRadiation: solarRadiation,
+		windSpeed: historicData.daily.data[ 0 ].windSpeed,
+		// TODO find out what height wind speed measurements are actually taken at.
+		windSpeedMeasurementHeight: 2,
+		dayOfYear: moment().subtract( 1, "day" ).dayOfYear(),
+		lat: coordinates[ 0 ],
+		precip: historicData.daily.data[ 0 ].precipIntensity * 24
```

Comment:
If we consolidate this into WateringData then suggest adding `lon` as well for completeness.

Reply:
On second thoughts, `local` wont be able to supply lat/lon informtion. Suggest we pass `coordinates` to adjustment methods instead.

Reply:
It looks like it should be possible to remove `lat` from `EToData` and just pass `coordinates` to `calculateETo` instead, but I think that intentionally avoided doing this for some reason that I can't currently remember. If I can't figure out what that reason was (which would indicate that the original reasoning is probably no longer applicable), I'll remove this field from `EToData` since it's excessive to have it there.